common/bin/oo-diagnostics

```diff
@@ -1226,6 +1228,28 @@ class OODiag
       do_warn "Using a self-signed certificate for the broker"
     end
 
+    # Retrieve the SSL cert from broker_host_env
+    env_response = `curl -k -s -v https://#{broker_host_env.chomp}/ -o /dev/null 2>&1`
+    if $? != 0
+      do_fail <<-CANNOTCONNECT
+        An error arose while trying to connect to the SSL reverse proxy.  curl returned with
+        #{env_response.strip.empty? ? "no output." : "the following output:\n" + env_response}
+
+        Please check whether the httpd service is running.
+      CANNOTCONNECT
+      return
+    end
+
+    # Get ssl cert's issuer, subject and commonname
+    env_issuer = env_response.slice(/^\*\s*issuer: .*$/).gsub(/^\*\s*issuer: /,"")
+    env_subject = env_response.slice(/^\*\s*subject: .*$/).gsub(/^\*\s*subject: /,"")
+    env_commonname = env_response.slice(/^\*\s*common name: .*$/).gsub(/^\*\s*common name: /,"")
+
+    # does ssl cert from broker_host_env matching the current broker
+    do_fail("SSL cert issuer doesn't match between current broker and broker host") if (env_issuer != issuer)
+    do_fail("SSL cert subject doesn't match between current broker and broker host") if (env_subject != subject)
+    do_fail("SSL cert commonname doesn't match between current broker and broker host") if (env_commonname != commonname)
```

Comment:
These error messages are likely not very helpful to an administrator. The admin would not know the difference between 'current broker' and 'broker host'.

Perhaps something along the lines of:
```
do_fail("SSL cert issuer doesn't match between localhost and #{broker_host_env} defined in /etc/openshift/env/OPENSHIFT_BROKER_HOST")
```